lib/ts/nextjs.ts

```diff
@@ -127,6 +128,141 @@ export default class NextJS {
             });
         };
     }
+
+    private static async commonSSRSession(
+        baseRequest: PreParsedRequest,
+        options?: VerifySessionOptions
+    ): Promise<{
+        session: SessionContainer | undefined;
+        hasToken: boolean;
+        hasInvalidClaims: boolean;
+        baseResponse: CollectingResponse;
+        nextResponse?: Response;
+    }> {
+        let baseResponse = new CollectingResponse();
+
+        try {
+            let session = await Session.getSession(baseRequest, baseResponse, options);
+            return {
+                session,
+                hasInvalidClaims: false,
+                hasToken: session !== undefined,
+                baseResponse,
+            };
+        } catch (err) {
+            if (Session.Error.isErrorFromSuperTokens(err)) {
+                return {
+                    hasToken: err.type !== Session.Error.UNAUTHORISED,
+                    hasInvalidClaims: err.type === Session.Error.INVALID_CLAIMS,
+                    session: undefined,
+                    baseResponse,
+                    nextResponse: new Response("Authentication required", {
```

Comment:
The Response class being used here is not the nextjs response class - so how is this working?

Reply:
[NextResponse](https://nextjs.org/docs/app/api-reference/functions/next-response) extends Response and just adds some utility functions on it. It is okay to do this as we are returning the response. 

I thought about taking `NextResponse` as input but didn't really feel the need for it.

Reply:
Okay, please test this in an actual app and make sure that try refresh token works as expected.